### ISSUE TITLE
Feature Request: an initializer with a single size for the entire HexGrid

### ISSUE THREAD

**Issue by mgrider** [fananek/hex-grid#8]
**Problem**
When drawing a grid, it would be nice to "fit" the grid inside a specific `CGSize`.

**Ideal solution**
Something like:
```
let hexGrid = HexGrid(shape: GridShape.hexagon(2), gridSize: CGSize(width: 200, height: 200))
```

**Describe alternatives you've considered** 
Currently, if I want to "fit" the grid to a size, I need to do the math to figure out how many across and divide the total width by that number, do the same for height, and then I take the smaller number and use that. The new initializer could do something similar, but I'm not even sure how to do that for all the possible grid types, so it might also be nice to have functions that return the max number of hexes horizontally, or vertically, and possibly get the hex "farthest" in a given direction, maybe?

**Comment by fananek** [comment]
Hi, thanks for bringing up the idea. Let me think about it a bit.
We can't use any CoreGraphics components like `CGSize`, because the library is designed to be platform independent. On the other hand, it should not be that hard to introduce some abstract representation of Size (similar to `Point`).

BTW: Any PR is more than welcome.

**Comment by mgrider** [comment]
I'm not sure whether I need the `.rectangle` case for my project(s), but I've already created the following in my project:

```Swift
import CoreGraphics
import HexGrid

extension HexGrid {

    /// Convenience initializer that takes a `gridSize` parameter and computes `hexSize` from that.
    convenience init(
        shape: GridShape,
        gridSize: CGSize = CGSize(width: 100.0, height: 100.0),
        orientation: Orientation = Orientation.pointyOnTop,
        offsetLayout: OffsetLayout = OffsetLayout.even,
        origin: Point = Point(x: 0, y: 0),
        attributes: [String: Attribute] = [String: Attribute]()
    ) {
        var hexWidth: Double = 10
        var hexHeight: Double = 10
        switch shape {
        case .rectangle(let width, let height):
            break
        case .hexagon(let side):
            let totalAcross = (CGFloat(side) * 2.0) + 1.0
            switch orientation {
            case .pointyOnTop:
                hexWidth = gridSize.width / totalAcross
                hexHeight = hexWidth / 3.squareRoot()
                hexWidth = hexHeight
            case .flatOnTop:
                hexHeight = gridSize.height / totalAcross
                hexWidth = hexHeight / 3.squareRoot()
                hexHeight = hexWidth
            }
        case .triangle(let dimension):
            let totalAcross = CGFloat(dimension) + 1.0
            switch orientation {
            case .pointyOnTop:
                hexWidth = gridSize.width / totalAcross
                hexHeight = hexWidth / 3.squareRoot()
                hexWidth = hexHeight
            case .flatOnTop:
                hexHeight = gridSize.height / totalAcross
                hexWidth = hexHeight / 3.squareRoot()
                hexHeight = hexWidth
            }
        }
        self.init(
            shape: shape,
            orientation: orientation,
            offsetLayout: offsetLayout,
            hexSize: HexSize(width: hexWidth, height: hexHeight),
            origin: origin,
            attributes: attributes)
    }
}
```
If I end up filling out the rectangle case, I'll definitely submit a PR. BTW, I'm also using `SpriteKit` for drawing, as opposed to `SwiftUI`. (I found the example project definitely has some issues I didn't want to work around. In the simulator anyway.) It's currently not doing much more than drawing some hexagons, so maybe I'll push it up as another example/demo.

I think it might be nice to have a companion Package that is all about drawing hexagons. The lines are kind of blurry here, what with `origin` and `pixelCoordinates`, etc.

**Comment by mgrider** [comment]
I just posted my example project here: https://github.com/mgrider/SKHexGrid

I included my current todo list in the readme.

**Comment by fananek** [comment]
Awesome, thanks! I added link to your demo into read.me

**Comment by mgrider** [comment]
Considering this a bit more, I wonder if a better solution is to ignore the `GridShape` entirely, and instead wait until the existing `init` functions have already found a total size (the `pixelWidth` and `pixelHeight` in `updatePixelDimensions`), then resize everything based on the difference between the desired size and what's already been computed.

This would require modifications to the library, as opposed to just a new extension, but would have the added benefit of supporting custom grid arrangements as well. I might play with that a bit today.

**Comment by mgrider** [comment]
This is addressed in PR #12.
